### PR TITLE
feat(core): implement datetime formatter pipes

### LINKED ISSUE #5499: Add date formatting pipes to ngx core that rely on DatetimeAdapter
fundamental-ngx should provide date and datetime formatting pipes that allow the developer to quickly format dates in the template.  The pipes should depend on DatetimeAdapter to favor formatting in whatever date type the developer is using whether its FdDatetimeAdapter or the moment implementation etc.

### PR DESCRIPTION
**Pull request by mikerodonnell89**
fixes #5499 

#### Please provide a brief summary of this pull request.

Introduces 3 new pipes for formatting dates based up on the dates provided to `DATE_TIME_FORMATS` by the developer.

`dateFormatPipe` formats dates
`dateTimeFormatPipe` formats datetimes
`dateFromNowPipe` can potentially be used to display how much time has passed since a date - this is currently not implemented with the default DATE_TIME_FORMATS/FdDateTimeAdapter but will be easy to add to the adapter when using moment.js or some similar library.

This does introduce a breaking change. Optional `fromNow` function was added to the DatetimeAdapter abstract class.  Because there are no truly "optional" abstract functions, any adapter that implements the DatetimeAdapter will need to implement the `fromNow` function - however if the developer does not wish to use the `fromNow` function, in their custom adapter they can simply implement it as `fromNow: undefined` as I've done here for the default FdDatetimeAdapter

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/utils/pipes/dateFormat.pipes.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@Pipe({
+    name: 'dateFormatPipe'
+})
+export class DateFormatPipe<D> implements PipeTransform {
+    constructor(
+        private _dateTimeAdapter: DatetimeAdapter<D>,
+        @Optional() @Inject(DATE_TIME_FORMATS) private _dateTimeFormats: DateTimeFormats
+    ) {}
+
+    transform(date: D): any {
```
Comment (Betrozov) [inline libs/core/src/lib/utils/pipes/dateFormat.pipes.ts:14]: should we change any to string?
as I see `this._dateTimeAdapter.format` returns `string`

File: libs/core/src/lib/utils/pipes/dateFormat.pipes.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+@Pipe({
+    name: 'dateTimeFormatPipe'
+})
+export class DateTimeFormatPipe<D> implements PipeTransform {
+    constructor(
+        private _dateTimeAdapter: DatetimeAdapter<D>,
+        @Optional() @Inject(DATE_TIME_FORMATS) private _dateTimeFormats: DateTimeFormats
+    ) {}
+
+    transform(date: D): any {
```
Comment (Betrozov) [inline libs/core/src/lib/utils/pipes/dateFormat.pipes.ts:28]: same

File: libs/core/src/lib/utils/pipes/dateFormat.pipes.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+@Pipe({
+    name: 'dateFromNowPipe'
+})
+export class DateFromNowPipe<D> implements PipeTransform {
+    constructor(
+        private _dateTimeAdapter: DatetimeAdapter<D>
+    ) {}
+
+    transform(date: D): any {
```
Comment (Betrozov) [inline libs/core/src/lib/utils/pipes/dateFormat.pipes.ts:41]: same

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Comment by droshev** [comment]
Closing old PRs - feel free to reopen it if/when needed

File: apps/docs/src/app/core/component-docs/date-picker/examples/date-picker-format-example.component.ts [hunk 4]
```diff
@@ -25,12 +25,12 @@ export const CUSTOM_FD_DATETIME_FORMATS: DateTimeFormats = {
     template: `
         <fd-date-picker [(ngModel)]="date" placeholder="dd-mm-yyyy"></fd-date-picker>
         <br />
-        <div>Selected Date: {{ date?.toDateString() }}</div>
+        <div>Selected Date: {{ date | dateFormatPipe }}</div>
```
Comment (N1XUS) [inline apps/docs/src/app/core/component-docs/date-picker/examples/date-picker-format-example.component.ts:28]: According to the other pipelines, and angular's naming convention, it's better to call the pipe `dateFormat`

**Review by N1XUS — CHANGES_REQUESTED** [review]

File: libs/core/src/lib/utils/pipes/dateFormat.pipes.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { Inject, Optional, Pipe, PipeTransform } from '@angular/core';
+import { DATE_TIME_FORMATS, DateTimeFormats } from '../../datetime/datetime-formats';
+import { DatetimeAdapter } from '../../datetime/datetime-adapter';
```
Comment (N1XUS) [inline libs/core/src/lib/utils/pipes/dateFormat.pipes.ts:3]: Import from the correct namespace

File: apps/docs/src/app/core/component-docs/datetime-picker/examples/datetime-format-example/datetime-format-example.component.html [hunk 6]
```diff
@@ -5,4 +5,4 @@
 <br />
 <br />
 <br />
-Selected: {{ date || 'null' }}
\ No newline at end of file
+Selected: {{ date ? (date | dateTimeFormatPipe) : 'null' }}
```
Comment (N1XUS) [inline apps/docs/src/app/core/component-docs/datetime-picker/examples/datetime-format-example/datetime-format-example.component.html:8]: You can use optional chaining for this, and cover the scenario in the pipe

**Review by N1XUS — CHANGES_REQUESTED** [review]

**Comment by N1XUS** [comment]
@mikerodonnell89 please fix Travis CI failing build

**Review by Betrozov — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/datetime-picker/examples/datetime-format-example/datetime-format-example.component.html (hunk 6 above)
Comment (mikerodonnell89) [reply apps/docs/src/app/core/component-docs/datetime-picker/examples/datetime-format-example/datetime-format-example.component.html:8]: In my opinion it is better to allow the application developer to handle this themselves, maybe they want to display "null" or maybe "no date provided" etc.

**Review by N1XUS — APPROVED** [review]

**Pull request merged**